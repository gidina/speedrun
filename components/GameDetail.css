.game-detail {
    grid-template-columns: 50% 50%;
    grid-template-rows: 25% 25% 25% 25%;
    grid-column-gap: 10px;
    grid-row-gap: 10px;
    display: grid;
    align-items: center;
    justify-items: center;
    width: 50%;
    font-family: Arial, Helvetica, sans-serif;
}

.game-detail-img {
    width: 100%;
    grid-column-start: 1;
    grid-column-end: 2;
    grid-row-start: 1;
    grid-row-end: 5;
}

.game-detail-name {
    grid-column-start: 2;
    grid-column-end: 3;
    grid-row-start: 1;
    grid-row-end: 2;
    font-weight: bold;
}

.game-detail-player {
    grid-column-start: 2;
    grid-column-end: 3;
    grid-row-start: 2;
    grid-row-end: 3;
}

.game-detail-runs {
    grid-column-start: 2;
    grid-column-end: 3;
    grid-row-start: 3;
    grid-row-end: 4;
}

.game-detail-video {
    grid-column-start: 2;
    grid-column-end: 4;
    grid-row-start: 4;
    grid-row-end: 5;

    padding: 6px 20px 6px 20px;
    background-color: #e7e7e7;
    border-radius: 3px;
}

.game-detail-video a {
    text-decoration: none;
    color: black;
}

.game-detail-video:hover {
    border-color: black;
    cursor: pointer;
    font-weight: bold;
}