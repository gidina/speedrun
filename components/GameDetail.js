import React from "react";
import PropTypes from 'prop-types';

import "./game.css";

const GameDetail = props => {
    const { isLoadingRuns, selectedGame } = props;
    
    if (isLoadingRuns) {
        return <div>Loading...</div>;
    }

    if (!selectedGame) {
        return <div>Ooops, this game does not exist</div>;
    }

    const { name, logoUrl, firstRun } = selectedGame;
    const { firstPlayerName, timePlayed, videoUrl } = firstRun;

    return <div className="game-detail">
        <img className="game-detail-img" alt="" src={logoUrl} />
        <span className="game-detail-name">{name}</span>
        <div className="game-detail-player">{firstPlayerName}</div>
        <span className="game-detail-runs">{timePlayed}</span>
        <button className="game-detail-video"><a href={videoUrl}>Video</a></button>
    </div>
};

export default GameDetail;

GameDetail.propTypes = {
    selectedGame: PropTypes.shape({
        name: PropTypes.string,
        logoUrl: PropTypes.string,
        firstRun: PropTypes.shape({
            firstPlayerName: PropTypes.string,
            timePlayed: PropTypes.integer,
            videoUrl: PropTypes.string
        })
    })
};