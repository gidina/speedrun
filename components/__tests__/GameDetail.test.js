import React from 'react';
import renderer from 'react-test-renderer';
import GameDetail from '../GameDetail';

it('Renders correctly', () => {
    const selectedGame = {
        name: "selected-game-name",
        logoUrl: "selected-game-logo-url",
        firstRun: {
            players: [
                {
                    id: "first-player-id"
                }
            ],
            videos: {
                links: [
                    {
                        uri: "selected-game-video-url"
                    }
                ]
            },
            times: {
                primary_t: 493
            }
        }
    };
    const tree = renderer
        .create(<GameDetail selectedGame={selectedGame} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("If firstRun doesn't exist on selectedGame prop it renders null", () => {
    const selectedGame = {
        name: "selected-game-name",
        logoUrl: "selected-game-logo-url"
    };
    const tree = renderer
        .create(<GameDetail selectedGame={selectedGame} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("If first player doesn't exist on selectedGame.firstRun prop it renders null", () => {
    const selectedGame = {
        name: "selected-game-name",
        logoUrl: "selected-game-logo-url",
        firstRun: {
            videos: {
                links: [
                    {
                        uri: "selected-game-video-url"
                    }
                ]
            },
            times: {
                primary_t: 493
            }
        }
    };
    const tree = renderer
        .create(<GameDetail selectedGame={selectedGame} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});