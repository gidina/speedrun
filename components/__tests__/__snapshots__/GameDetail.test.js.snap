// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`If first player doesn't exist on selectedGame.firstRun prop it renders null 1`] = `null`;

exports[`If firstRun doesn't exist on selectedGame prop it renders null 1`] = `null`;

exports[`Renders correctly 1`] = `
<div
  className="game-detail"
>
  <img
    alt=""
    className="game-detail-img"
    src="selected-game-logo-url"
  />
  <span
    className="game-detail-name"
  >
    selected-game-name
  </span>
  <div
    className="game-detail-player"
  >
    first-player-id
  </div>
  <span
    className="game-detail-runs"
  >
    493
  </span>
  <button
    className="game-detail-video"
  >
    <a
      href="selected-game-video-url"
    >
      Video
    </a>
  </button>
</div>
`;
